src/app/services/selection.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+
+import { Section } from '../course-list/section/section';
+import { Course } from '../course-list/course/course';
+
+@Injectable()
+export class SelectionService {
+
+  public toggleSection(section : Section) {
+    this.isSectionSelected(section) ? this.removeSection(section) : this.addSection(section);
+  }
+
+  public addSection(section : Section) {
+    let store = JSON.parse(localStorage.getItem('selections')) || {};
+    store[section.course_id] = store[section.course_id] || [];
+    if (store[section.course_id].includes(section.id)) return;
+    store[section.course_id].push(section.id);
+    store[section.course_id].sort();
+    localStorage.setItem('selections', JSON.stringify(store));
+  }
+
+  public removeSection(section : Section) {
+    let store = JSON.parse(localStorage.getItem('selections'));
```

Comment:
do `let store = JSON.parse(localStorage.getItem('selections')) || {};` here as well for consistency and remove !store from next line.